moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js

```diff
@@ -96,139 +109,67 @@ const ProjectSummaryKnackDataTrackerSync = ({
     Object.keys(fieldMap).forEach(element => {
       body[element] = project[fieldMap[element]];
     });
-
-    if (signalIds.length > 0) {
-      // REACT_APP_KNACK_DATA_TRACKER_SIGNAL_CONNECTION contains the signalId connection field to the signals table
-      body[
-        process.env.REACT_APP_KNACK_DATA_TRACKER_SIGNAL_CONNECTION
-      ] = signalIds;
-    }
-
+    // REACT_APP_KNACK_DATA_TRACKER_SIGNAL_CONNECTION contains the signalId connection field to the signals table
+    // it's ok if the signalIds array is empty
+    body[
+      process.env.REACT_APP_KNACK_DATA_TRACKER_SIGNAL_CONNECTION
+    ] = signalIds;
     return JSON.stringify(body);
   };
 
-  const numberOfSignalsInProject = countSignalsInProject(project);
-
   /**
    * Function to handle the actual mechanics of synchronizing the data on hand to the Knack API endpoint.
    */
   const handleSync = () => {
-    if (project.knack_project_id) {
-      // The following code, contained in this block, handles a "re-sync" to knack for a given project.
-      // Currently, our /UI does not contain an element that allows a user to request a re-sync, but
-      // consensus is that this code may be useful enough in the future to warrant leaving it in-place.
-
-      // updating knack record
-      fetch(knackProjectEndpointUrl, {
-        // Fetch will return a promise, allowing us to start a chain of .then() calls
-        method: "GET",
-        headers: buildHeaders,
+    // The following code is capabale of handling a "re-sync" to knack for a given project.
+    // Currently, our /UI does not contain an element that allows a user to request a re-sync, but
+    // consensus code is ready to "re-sync" a project thanks to it's use of a dynamic knackHttpMethod
+    console.log("HTTP method: " + knackHttpMethod);
+    // POST (create) or PUT (update) a project record in Knack
+    const signalIds = signals.map(signal => signal.knack_id);
+    return fetch(knackProjectEndpointUrl, {
+      method: knackHttpMethod,
+      headers: buildHeaders,
+      body: buildBody(signalIds),
+    })
```

Comment:
just using a single `fetch` call which pushes the current state of the moped project to Knack, either updating it or creating it in Knack based on `knackHttpMethod`.